### PR TITLE
Apply `black` to remaining parts of `astropy`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to remaining parts of `astropy` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

When this PR is merged along with all the other black PRs, only `examples`, `extern` and a few automatically generated files will be excluded from `black` formatting in `astropy.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by astrofrog** [comment]
Looks good to me, needs a rebase and ready for string normalization

File: docs/conf.py [hunk 1]
```diff
@@ -93,58 +95,88 @@
...
-intersphinx_mapping['pyerfa'] = ('https://pyerfa.readthedocs.io/en/stable/', None)  # noqa: F405
-intersphinx_mapping['pytest'] = ('https://docs.pytest.org/en/stable/', None)  # noqa: F405
-intersphinx_mapping['ipython'] = ('https://ipython.readthedocs.io/en/stable/', None)  # noqa: F405
-intersphinx_mapping['pandas'] = ('https://pandas.pydata.org/pandas-docs/stable/', None)  # noqa: E501, F405
-intersphinx_mapping['sphinx_automodapi'] = ('https://sphinx-automodapi.readthedocs.io/en/stable/', None)  # noqa: E501, F405
-intersphinx_mapping['packagetemplate'] = ('https://docs.astropy.org/projects/package-template/en/latest/', None)  # noqa: E501, F405
-intersphinx_mapping['h5py'] = ('https://docs.h5py.org/en/stable/', None)  # noqa: F405
-intersphinx_mapping['asdf-astropy'] = ('https://asdf-astropy.readthedocs.io/en/latest/', None)  # noqa: F405
-intersphinx_mapping['fsspec'] = ('https://filesystem-spec.readthedocs.io/en/latest/', None)  # noqa: F405
+intersphinx_mapping["astropy-dev"] = (  # noqa: F405
```
Comment (pllim) [inline docs/conf.py:101]: I don't like the `intersphinx_mapping` changes. You cannot grep it and look at a certain package mapping anymore without opening the file. But maybe it is just me?

Comment (WilliamJamieson) [reply docs/conf.py:101]: I'll revert this to the original and add `# fmt: off`/`# fmt: on`.

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

**Comment by WilliamJamieson** [comment]
Can we make sure this is the last apply `black` PR to be merged?

**Pull request merged**